src/NetMQ/Core/Patterns/XPub.cs

```diff
@@ -176,15 +184,29 @@ protected override void XReadActivated(Pipe pipe)
                         // If the subscription is not a duplicate, store it so that it can be
                         // passed to used on next recv call.
                         if (m_options.SocketType == ZmqSocketType.Xpub && (unique || m_verbose))
-                            m_pending.Enqueue(sub.CloneData());
+                        {
+                            m_pending.Enqueue(sub);
+                        }
+                        else
+                        {
+                            sub.Close();
+                        }
+
                     }
                 }
+                else if (m_broadcastEnabled && size > 0 && sub.HasMore && sub[0] == 2)
+                {
+                    m_lastPipe = pipe;
+                    m_lastPipeIsBroadcast = true;
+                    sub.Offset = sub.Offset + 1;
+                    sub.Size = sub.Size - 1;
+                    m_pending.Enqueue(sub);
```

Comment:
PubSub can also work with one frame only... the first part being a topic is not in the code or RFC. just convenient.

Anyway I'm talking about the XPub receive...

So before this change, if you call XPub receive you can get:
If it a subscribe it will be prefix with 0, ubsubscribe with 1 and the rest with something else

Now you actually get a message that you cannot tell what type is it, because you removed the prefix.

The user code that call for Send should remove the prefix.